File: backend/app/Models/Route.php
<?php

declare(strict_types=1);

namespace App\Models;

use App\Concerns\HasUuid;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\SoftDeletes;

class Route extends Model
{
    use HasUuid, SoftDeletes;

    protected $table = 'routes';

    protected $fillable = [
        'code', 'origin_hub_id', 'destination_hub_id', 'display_name',
        'estimated_duration_minutes', 'is_active',
    ];

    protected function casts(): array
    {
        return ['is_active' => 'boolean', 'estimated_duration_minutes' => 'integer'];
    }

    public function originHub(): BelongsTo
    {
        return $this->belongsTo(Hub::class, 'origin_hub_id');
    }

    public function destinationHub(): BelongsTo
    {
        return $this->belongsTo(Hub::class, 'destination_hub_id');
    }

    public function trips(): HasMany
    {
        return $this->hasMany(Trip::class);
    }

    public function pricingMatrix(): HasMany
    {
        return $this->hasMany(PricingMatrix::class);
    }
}

File: mobile_sender/lib/models/route.dart
class Route {
  final String id;
  final String code;
  final String originHub;
  final String destinationHub;
  final int distance;
  final bool isActive;
  final Map<String, int> pricingMatrix; // size -> price

  Route({
    required this.id,
    required this.code,
    required this.originHub,
    required this.destinationHub,
    required this.distance,
    required this.isActive,
    required this.pricingMatrix,
  });

  factory Route.fromJson(Map<String, dynamic> json) {
    return Route(
      id: json['id'],
      code: json['code'],
      originHub: json['origin_hub'],
      destinationHub: json['destination_hub'],
      distance: json['distance'],
      isActive: json['is_active'],
      pricingMatrix: Map<String, int>.from(json['pricing_matrix'] ?? {}),
    );
  }

  String get displayName => '$originHub ↔ $destinationHub';

  int getPriceForSize(String size) {
    return pricingMatrix[size] ?? 0;
  }
}
